### PR TITLE
Fixing linter hints automatically.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

After updating ESlint and its related configuration packages, an `eslint
--fix` run was performed.

The changes were triggered by the following rules (sorted by count):

```
{ 'react/self-closing-comp': 1,
  'no-lonely-if': 1,
  'key-spacing': 2,
  'prefer-template': 2,
  'space-before-function-paren': 2,
  'prefer-const': 2,
  'padded-blocks': 2,
  'comma-dangle': 3,
  'one-var': 3,
  semi: 3,
  'implicit-arrow-linebreak': 3,
  'eol-last': 3,
  'no-confusing-arrow': 4,
  'keyword-spacing': 6,
  'one-var-declaration-per-line': 7,
  'quote-props': 8,
  'object-curly-newline': 8,
  'no-else-return': 9,
  'no-multiple-empty-lines': 12,
  eqeqeq: 13,
  'arrow-parens': 14,
  'lines-between-class-members': 17,
  'react/jsx-indent': 25,
  'dot-notation': 27,
  'import/order': 32,
  'react/jsx-tag-spacing': 32,
  'object-curly-spacing': 34,
  'react/jsx-curly-brace-presence': 45,
  indent: 51,
  'prefer-arrow-callback': 54,
  quotes: 58,
  'react/jsx-closing-bracket-location': 58,
  'react/jsx-first-prop-new-line': 70,
  'function-paren-newline': 80,
  'react/jsx-closing-tag-location': 140,
  'no-var': 149,
  'react/jsx-indent-props': 157,
  'operator-linebreak': 162,
  'react/jsx-wrap-multilines': 165,
  'import/newline-after-import': 214,
  'import/first': 240,
  'prefer-destructuring': 250,
  'react/jsx-max-props-per-line': 296 }
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: graylog2-web-interface/src/components/alertconditions/AlertCondition.jsx [hunk 1]
```diff
@@ -57,15 +57,15 @@ const AlertCondition = createReactClass({
   },
 
   render() {
-    const stream = this.props.stream;
+    const { stream } = this.props;
     const condition = this.props.alertCondition;
-    const conditionType = this.props.conditionType;
+    const { conditionType } = this.props;
```
[the remark this answers is not in the document]
Comment (kyleknighted) [reply graylog2-web-interface/src/components/alertconditions/AlertCondition.jsx]: But not sure why the rename of the variable - perhaps we keep it `alertCondition`?

File: graylog2-web-interface/src/components/alertconditions/AlertConditionForm.jsx [hunk 2]
```diff
@@ -59,8 +59,8 @@ const AlertConditionForm = createReactClass({
   },
 
   render() {
-    const alertCondition = this.props.alertCondition;
-    const conditionType = this.props.conditionType;
+    const { alertCondition } = this.props;
+    const { conditionType } = this.props;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/alertconditions/AlertConditionForm.jsx]: Could be merged into one.

File: graylog2-web-interface/src/components/alertconditions/AlertConditionSummary.jsx [hunk 3]
```diff
@@ -25,14 +25,14 @@ class AlertConditionSummary extends React.Component {
   };
 
   render() {
-    const stream = this.props.stream;
+    const { stream } = this.props;
     const condition = this.props.alertCondition;
-    const conditionType = this.props.conditionType;
+    const { conditionType } = this.props;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/alertconditions/AlertConditionSummary.jsx]: These three lines could be merged

File: graylog2-web-interface/src/components/alertconditions/UnknownAlertCondition.jsx [hunk 4]
```diff
@@ -13,7 +13,7 @@ class UnknownAlertCondition extends React.Component {
 
   render() {
     const condition = this.props.alertCondition;
-    const stream = this.props.stream;
+    const { stream } = this.props;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/alertconditions/UnknownAlertCondition.jsx]: These two lines could be merged.

File: graylog2-web-interface/src/components/alertconditions/AlertConditionSummary.jsx (hunk 3 above)
Comment (kyleknighted) [reply graylog2-web-interface/src/components/alertconditions/AlertConditionSummary.jsx]: Not sure about the rename of `alertCondition`

File: graylog2-web-interface/src/components/alertconditions/fieldcontentcondition/FieldContentConditionSummary.jsx [hunk 5]
```diff
@@ -15,9 +15,9 @@ class FieldContentConditionSummary extends React.Component {
   };
 
   render() {
-    const alertCondition = this.props.alertCondition;
-    const field = alertCondition.parameters.field;
-    const value = alertCondition.parameters.value;
+    const { alertCondition } = this.props;
+    const { field } = alertCondition.parameters;
+    const { value } = alertCondition.parameters;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/alertconditions/fieldcontentcondition/FieldContentConditionSummary.jsx]: field and value could be merged.

File: graylog2-web-interface/src/components/alertconditions/fieldvaluecondition/FieldValueConditionSummary.jsx [hunk 6]
```diff
@@ -12,12 +12,12 @@ class FieldValueConditionSummary extends React.Component {
...
-    const alertCondition = this.props.alertCondition;
-    const field = alertCondition.parameters.field;
-    const threshold = alertCondition.parameters.threshold;
+    const { alertCondition } = this.props;
+    const { field } = alertCondition.parameters;
+    const { threshold } = alertCondition.parameters;
     const thresholdType = alertCondition.parameters.threshold_type.toLowerCase();
     const type = alertCondition.parameters.type.toLowerCase();
-    const time = alertCondition.parameters.time;
+    const { time } = alertCondition.parameters;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/alertconditions/fieldvaluecondition/FieldValueConditionSummary.jsx]: field, threshold, and time could be merged

File: graylog2-web-interface/src/components/alertconditions/messagecountcondition/MessageCountConditionSummary.jsx [hunk 7]
```diff
@@ -12,16 +12,17 @@ class MessageCountConditionSummary extends React.Component {
   };
 
   render() {
-    const alertCondition = this.props.alertCondition;
-    const threshold = alertCondition.parameters.threshold;
+    const { alertCondition } = this.props;
+    const { threshold } = alertCondition.parameters;
     const thresholdType = alertCondition.parameters.threshold_type.toLowerCase();
-    const time = alertCondition.parameters.time;
+    const { time } = alertCondition.parameters;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/alertconditions/messagecountcondition/MessageCountConditionSummary.jsx]: threshold and time could be merged

File: graylog2-web-interface/src/components/alertnotifications/AlertNotification.jsx [hunk 8]
```diff
@@ -76,7 +76,7 @@ const AlertNotification = createReactClass({
     }
 
     const notification = this.props.alertNotification;
-    const stream = this.props.stream;
+    const { stream } = this.props;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/alertnotifications/AlertNotification.jsx]:
```js
const { alertNotification : notification, stream } = this.props;
```

Not sure if we need to rename `alertNotification` though

File: graylog2-web-interface/src/components/alerts/AlertDetails.jsx [hunk 9]
```diff
@@ -27,8 +28,8 @@ class AlertDetails extends React.Component {
   };
 
   render() {
-    const alert = this.props.alert;
-    const stream = this.props.stream;
+    const { alert } = this.props;
+    const { stream } = this.props;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/alerts/AlertDetails.jsx]: Could be merged

File: graylog2-web-interface/src/components/alerts/AlertTimeline.jsx [hunk 10]
```diff
@@ -98,7 +99,7 @@ const AlertTimeline = createReactClass({
       return <Spinner />;
     }
 
-    const alert = this.props.alert;
+    const { alert } = this.props;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/alerts/AlertTimeline.jsx]: condition and type could be merged in with this destruct and then do the conditionals below

File: graylog2-web-interface/src/components/configurationforms/BooleanField.jsx [hunk 11]
```diff
@@ -15,9 +15,9 @@ class BooleanField extends React.Component {
   };
 
   render() {
-    const field = this.props.field;
-    const typeName = this.props.typeName;
-    const title = this.props.title;
+    const { field } = this.props;
+    const { typeName } = this.props;
+    const { title } = this.props;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/configurationforms/BooleanField.jsx]: These could be merged.

File: graylog2-web-interface/src/components/configurationforms/ConfigurationForm.jsx [hunk 12]
```diff
@@ -116,47 +116,89 @@ const ConfigurationForm = createReactClass({
...
     }
   },
 
   render() {
-    const typeName = this.props.typeName;
-    const title = this.props.title;
-    const helpBlock = this.props.helpBlock;
+    const { typeName } = this.props;
+    const { title } = this.props;
+    const { helpBlock } = this.props;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/configurationforms/ConfigurationForm.jsx]: Could be merged into single destruct

File: graylog2-web-interface/src/components/configurationforms/DropdownField.jsx [hunk 13]
```diff
@@ -44,12 +44,12 @@ class DropdownField extends React.Component {
...
 
   render() {
-    const field = this.state.field;
+    const { field } = this.state;
     const options = $.map(field.additional_info.values, this._formatOption);
     if (this.props.addPlaceholder) {
       options.unshift(this._formatOption(`Select ${field.human_name || this.state.title}`, '', true));
     }
-    const typeName = this.state.typeName;
+    const { typeName } = this.state;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/configurationforms/DropdownField.jsx]: Could be merged with line 47

File: graylog2-web-interface/src/components/configurationforms/DropdownField.jsx [hunk 14]
```diff
@@ -58,9 +58,13 @@ class DropdownField extends React.Component {
           {FieldHelpers.optionalMarker(field)}
         </label>
 
-        <select id={field.title} value={this.state.value}
-                className="input-xlarge form-control" onChange={this.handleChange}
-                autoFocus={this.props.autoFocus} disabled={this.props.disabled} required={!field.is_optional}>
+        <select id={field.title}
+                value={this.state.value}
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/configurationforms/DropdownField.jsx]: destruct value with the rest

File: graylog2-web-interface/src/components/configurationforms/DropdownField.jsx [hunk 15]
```diff
@@ -58,9 +58,13 @@ class DropdownField extends React.Component {
...
+                className="input-xlarge form-control"
+                onChange={this.handleChange}
+                autoFocus={this.props.autoFocus}
+                disabled={this.props.disabled}
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/configurationforms/DropdownField.jsx]: Could destruct props with state

File: graylog2-web-interface/src/components/configurationforms/ListField.jsx [hunk 16]
```diff
@@ -42,9 +42,9 @@ class ListField extends React.Component {
   };
 
   render() {
-    const field = this.state.field;
-    const typeName = this.state.typeName;
-    const value = this.state.value;
+    const { field } = this.state;
+    const { typeName } = this.state;
+    const { value } = this.state;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/configurationforms/ListField.jsx]: Single destruct statement

File: graylog2-web-interface/src/components/configurationforms/NumberField.jsx [hunk 17]
```diff
@@ -50,8 +50,8 @@ const NumberField = createReactClass({
   },
 
   render() {
-    const typeName = this.props.typeName;
-    const field = this.props.field;
+    const { typeName } = this.props;
+    const { field } = this.props;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/configurationforms/NumberField.jsx]: Single destruct statement

File: graylog2-web-interface/src/components/configurationforms/NumberField.jsx [hunk 18]
```diff
@@ -63,9 +63,14 @@ const NumberField = createReactClass({
...
-               value={this.props.value} className="input-xlarge validatable form-control"
-          {...validationSpecs} autoFocus={this.props.autoFocus} />
+        <input id={field.title}
+               type="number"
+               required={isRequired}
+               onChange={this.handleChange}
+               value={this.props.value}
+               className="input-xlarge validatable form-control"
+               {...validationSpecs}
+               autoFocus={this.props.autoFocus} />
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/configurationforms/NumberField.jsx]: add to destruct

File: graylog2-web-interface/src/components/configurationforms/NumberField.jsx [hunk 19]
```diff
@@ -63,9 +63,14 @@ const NumberField = createReactClass({
...
         </label>
 
-        <input id={field.title} type="number" required={isRequired} onChange={this.handleChange}
-               value={this.props.value} className="input-xlarge validatable form-control"
-          {...validationSpecs} autoFocus={this.props.autoFocus} />
+        <input id={field.title}
+               type="number"
+               required={isRequired}
+               onChange={this.handleChange}
+               value={this.props.value}
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/configurationforms/NumberField.jsx]: add to destruct

File: graylog2-web-interface/src/components/configurationforms/TextField.jsx [hunk 20]
```diff
@@ -29,24 +29,35 @@ class TextField extends React.Component {
   };
 
   render() {
-    const field = this.state.field;
-    const title = this.state.title;
-    const typeName = this.state.typeName;
+    const { field } = this.state;
+    const { title } = this.state;
+    const { typeName } = this.state;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/configurationforms/TextField.jsx]: Single destruct statement

File: graylog2-web-interface/src/components/configurationforms/TextField.jsx [hunk 21]
```diff
@@ -29,24 +29,35 @@ class TextField extends React.Component {
...
-                  name={`configuration[${title}]`} required={isRequired} value={this.state.value}
-                  onChange={this.handleChange} autoFocus={this.props.autoFocus} />
+        <textarea id={title}
+                  className="form-control"
+                  rows={10}
+                  name={`configuration[${title}]`}
+                  required={isRequired}
+                  value={this.state.value}
+                  onChange={this.handleChange}
+                  autoFocus={this.props.autoFocus} />
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/configurationforms/TextField.jsx]: add to destruct statement

File: graylog2-web-interface/src/components/configurationforms/TextField.jsx [hunk 22]
```diff
@@ -29,24 +29,35 @@ class TextField extends React.Component {
...
       );
     } else {
       inputField = (
-        <input id={title} type={fieldType} className="form-control" name={`configuration[${title}]`} value={this.state.value}
-               onChange={this.handleChange} required={isRequired} autoFocus={this.props.autoFocus} />
+        <input id={title}
+               type={fieldType}
+               className="form-control"
+               name={`configuration[${title}]`}
+               value={this.state.value}
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/configurationforms/TextField.jsx]: add to destruct

File: graylog2-web-interface/src/components/configurationforms/TitleField.jsx [hunk 23]
```diff
@@ -17,11 +17,16 @@ class TitleField extends React.Component {
...
     const titleField = { is_optional: false, attributes: [], human_name: 'Title', description: this.props.helpBlock };
     return (
-      <TextField key={`${typeName}-title`} typeName={typeName} title="title" field={titleField}
-                                 value={this.props.value} onChange={this.props.onChange} autoFocus />
+      <TextField key={`${typeName}-title`}
+                 typeName={typeName}
+                 title="title"
+                 field={titleField}
+                 value={this.props.value}
+                 onChange={this.props.onChange}
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/configurationforms/TitleField.jsx]: Add to destruct statement

File: graylog2-web-interface/src/components/content-packs/ContentPackDetails.jsx [hunk 24]
```diff
@@ -12,8 +12,8 @@ import 'components/content-packs/ContentPackDetails.css';
 
 const ContentPackDetails = (props) => {
   const markdownDescription = markdown.toHTML(props.contentPack.description || '');
-  const contentPack = props.contentPack;
-  const constraints = props.constraints;
+  const { contentPack } = props;
+  const { constraints } = props;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/content-packs/ContentPackDetails.jsx]: Single destruct statement

File: graylog2-web-interface/src/components/content-packs/ContentPackDetails.jsx [hunk 25]
```diff
@@ -33,30 +33,34 @@ const ContentPackDetails = (props) => {
...
           <div>
             <ContentPackConstraints constraints={constraints} />
             <br />
           </div>
+          )
           }
-          { contentPack.entities && contentPack.entities.length > 0 && props.verbose &&
-          <ContentPackEntitiesList contentPack={props.contentPack} readOnly />
+          { contentPack.entities && contentPack.entities.length > 0 && props.verbose
+          && <ContentPackEntitiesList contentPack={props.contentPack} readOnly />
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/content-packs/ContentPackDetails.jsx]: Use destructed var `contentPack`

File: graylog2-web-interface/src/components/content-packs/ContentPackDetails.jsx [hunk 26]
```diff
@@ -33,30 +33,34 @@ const ContentPackDetails = (props) => {
...
           }
-          { contentPack.parameters && contentPack.parameters.length > 0 && props.verbose &&
-          <ContentPackParameterList contentPack={props.contentPack} readOnly />
+          { contentPack.parameters && contentPack.parameters.length > 0 && props.verbose
+          && <ContentPackParameterList contentPack={props.contentPack} readOnly />
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/content-packs/ContentPackDetails.jsx]: Use destructed var `contentPack`

File: graylog2-web-interface/src/components/content-packs/ContentPackDetails.jsx [hunk 27]
```diff
@@ -33,30 +33,34 @@ const ContentPackDetails = (props) => {
...
+          && (
           <div>
             <ContentPackConstraints constraints={constraints} />
             <br />
           </div>
+          )
           }
-          { contentPack.entities && contentPack.entities.length > 0 && props.verbose &&
-          <ContentPackEntitiesList contentPack={props.contentPack} readOnly />
+          { contentPack.entities && contentPack.entities.length > 0 && props.verbose
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/content-packs/ContentPackDetails.jsx]: destruct `verbose`

File: graylog2-web-interface/src/components/content-packs/ContentPackEntityConfig.test.jsx [hunk 28]
```diff
@@ -14,11 +14,9 @@ describe('<ContentPackEntityConfig />', () => {
...
     const appliedParameter = [{ configKey: 'descr', paramName: 'descrParam' }];
     const parameter = [{ name: 'descrParam', title: 'A descr Parameter', type: 'string' }];
-    const wrapper = renderer.create(<ContentPackEntityConfig
-      entity={entity}
-      appliedParameter={appliedParameter}
-      parameters={parameter}
-    />);
+    const wrapper = renderer.create(<ContentPackEntityConfig entity={entity}
+                                                             appliedParameter={appliedParameter}
+                                                             parameters={parameter} />);
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/content-packs/ContentPackEntityConfig.test.jsx]: Don't recall exactly which eslint flag lines up components like this - but maybe we can discuss its usefulness vs just making sure props are tabbed in once?

**Comment by dennisoelkers** [comment]
@kyleknighted: We talked about the suboptimal destructuring fixes yesterday and considered the outcome to be better than before, while not being optimal. If we want to improve them manually, I would suggest doing it outside of the scope of this PR so the intention and origin (`eslint --fix` vs. manual) is clear.

**Comment by kyleknighted** [comment]
@dennisoelkers We can follow up after this PR with those fixes if necessary. Just think it would make more sense to get these stuff cleaned up once instead of having to continually go back time-and-time again. The automation got us 99% of the way there, some manual intervention will get us the rest of the way.

**Pull request merged**